crates/smt/src/smt.rs

```diff
@@ -0,0 +1,931 @@
+use std::collections::HashMap;
+
+use num_bigint::BigInt;
+
+use crate::utils::{
+    get_first_common_elements, get_index_of_last_non_zero_element, is_hexadecimal, key_to_path,
+};
+
+use std::fmt;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum Node {
+    Str(String),
+    BigInt(BigInt),
+}
```

Comment:
Yes, it might make sense to avoid doing these checks at runtime, but 90% of things can be represented by String and BigInt. I don't know if the game is worth the candle. Anyone has any additional comments here?